docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,825 @@
+    //
+    // +kubebuilder:validation:Required
+    // +required
+    // +unionDiscriminator
+    Topology NetworkTopology `json:"topology"`
+    ...
+    // Localnet is the Localnet topology configuration.
+    // +optional
+    Localnet *LocalnetConfig `json:"localnet,omitempty"`
+}
+```
+
+The CUDN spec should have additional validation rule for `spec.network.topology` field:
+```go
+// ClusterUserDefinedNetworkSpec defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkSpec struct {
+    ...
+    // +required
+    Network NetworkSpec `json:"network"`
+}
+```
+
+#### Localnet topology spec mutation support
+As mentioned in [Workflow Description](#workflow-description) section, it's possible to mutate the spec 
+of `NetworkAttachmentDefinition` (NAD), including for localnet topologies.
+The proposed solution should provider the similar experience when localnet topologies created using CUDN CRs. 
+
+As of today the CUDN has validation to ensure `spec.network` is immutable:
+```go
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Network spec is immutable"
+```
+This validation should be changed in a way it allows mutating localnet topology configurations,
+at least `MTU`, `VLAN`, `physicalNetworkName` and `excludeSubnets`.
+The proposed change is to invert the immutable-spec validation on CUDN `spec.network` as follows:
+1. Remove mutation restrictions on `spec.network`.
+2. Add CEL rule validations to restrict mutations of:
+   `spec.network.topology`, `spec.network.layer2`, and `spec.network.layer3`.
+3. Add CEL rule validations to restrict mutation of localnet topology settings:
+   `spec.network.localnet.role`, `spec.network.localnet.subnets` and `spec.network.topology.ipam`.
+
+#### Localnet topology configuration type
+Introduce new topology configuration type for localnet - `LocalnetConfig`.
+The CUDN CRD `spec.network` should feature proposed localnet topology configuration.
+
+The Layer2 and Layer3 configuration types (`Layer2Config` & `Layer3Config`) `role` field is defined with `NetworkRole` type.
+The `NetworkRole` type has the following enum validation, allowing `Secondary` and `Primary` values:
+```
+// +kubebuilder:validation:Enum=Primary;Secondary
+```
+The proposed localnet config type (`LocalnetConfig`) `role` field would have to accept `Secondary` value only.
+In order to avoid misleading CRD scheme, the enum validation should be moved closer to each `NetworkRole` usage:
+1. Remove the enum validation exist on `NetworkRole` definition
+2. At the `Layer2Config` definition, add enum validation to `role` field allowing: `Primary` or `Secondary`.
+3. At the `Layer3Config` definition, add enum validation to `role` field allowing: `Primary` or `Secondary`.
+4. At the proposed `LocalnetConfig` definition, add enum validation to `role` field allowing `Secondary` only.
+
+#### VLAN configurations
+
+As of today localnet topology allows setting VLAN for the network.
```

Comment:
you mean in the cni spec of the NAD definition right? Can you clarify that here please so that someone doesn't read this as the API you are suggesting for CUDN?